File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class OverviewResponse(BaseModel):
    assets_created_week: int
    assets_created_month: int
    total_remixes: int
    compliance_rate: float
    active_fscs: int


class TopRemixedItem(BaseModel):
    item_id: str
    artifact_name: str
    artifact_type: str
    product: str | None
    remix_count: int


class TopRemixedResponse(BaseModel):
    items: list[TopRemixedItem]


class ContentGap(BaseModel):
    product: str | None
    artifact_type: str
    fsc_count: int


class ContentGapResponse(BaseModel):
    gaps: list[ContentGap]


class ActivityDataPoint(BaseModel):
    date: str
    created: int
    exported: int


class ActivityResponse(BaseModel):
    data: list[ActivityDataPoint]
    granularity: str

File: frontend/src/types/analytics.ts
export interface OverviewMetrics {
  assets_created_week: number;
  assets_created_month: number;
  total_remixes: number;
  compliance_rate: number;
  active_fscs: number;
}

export interface TopRemixedItem {
  item_id: string;
  artifact_name: string;
  artifact_type: string;
  product: string | null;
  remix_count: number;
}

export interface TopRemixedResponse {
  items: TopRemixedItem[];
}

export interface ContentGap {
  product: string | null;
  artifact_type: string;
  fsc_count: number;
}

export interface ContentGapResponse {
  gaps: ContentGap[];
}

export interface ActivityDataPoint {
  date: string;
  created: number;
  exported: number;
}

export interface ActivityResponse {
  data: ActivityDataPoint[];
  granularity: string;
}
